purchase_order_report_grouped_by_vendor/views/report_purchase_order.xml

```diff
@@ -0,0 +1,145 @@
+                                        </td>
+                                        <td class="text-right">
+                                            <span t-field="line.product_qty" />
+                                            <span
+                                                t-field="line.product_uom.name"
+                                                groups="uom.group_uom"
+                                            />
+                                        </td>
+                                        <td class="text-right">
+                                            <span
+                                                t-field="line.date_planned"
+                                                t-options="{'widget': 'datetime', 'date_only': True, 'format': 'dd/MM/YY'}"
+                                            />
+                                        </td>
+                                        <td class="text-right">
+                                            <span t-field="line.price_unit" />
+                                        </td>
+                                        <td class="text-right">
+                                            <span
+                                                t-field="line.price_subtotal"
+                                                t-options='{"widget": "monetary", "display_currency": line.currency_id}'
+                                            />
+                                        </td>
+                                    </tr>
+                                </t>
+                            </t>
+                            <tr>
+                                <tr>
+                                    <td colspan="5" class="text-right">
+                                        <strong>Total:</strong>
+                                    </td>
+                                    <td class="text-right">
+                                        <span
+                                            t-esc="sum(line.price_subtotal for order in group['order_ids'] for line in order.order_line)"
+                                            t-options='{"widget": "monetary", "display_currency": order.currency_id}'
+                                        />
+                                    </td>
+                                </tr>
+                            </tr>
+                        </tbody>
+                    </table>
+                    <!-- Insert page break after each vendor -->
+                    <div style="page-break-after: always;" />
+                </div>
+            </t>
+        </t>
+    </template>
+
+    <template id="report_purchase_order_grouped_by_vendor">
+        <t t-call="web.html_container">
+            <t
+                t-call="purchase_order_report_grouped_by_vendor.report_purchase_order_grouped_by_vendor_document"
+                t-lang="docs[:1].partner_id.lang"
```

Comment:
The loop should be done here for each different partner and pass the corresponding lang